diagnostic_common_diagnostics/diagnostic_common_diagnostics/ntp_monitor.py

```diff
@@ -115,81 +77,135 @@ def __init__(self, ntp_hostname, offset=500, self_offset=500,
         self.self_stat.values = []
 
         self.mutex = threading.Lock()
-        self.pub = rospy.Publisher("/diagnostics", DIAG.DiagnosticArray, queue_size=10)
+        self.pub = self.create_publisher(DIAG.DiagnosticArray, "/diagnostics", 10)
 
         # we need to periodically republish this
         self.current_msg = None
-        self.pubtimer = rospy.Timer(rospy.Duration(.1), self.pubCB)
-        self.checktimer = rospy.Timer(rospy.Duration(.1), self.checkCB, True)
+        self.pubtimer = self.create_timer(0.1, self.pubCB)
+        self.checktimer = self.create_timer(0.1, self.checkCB)
 
-    def pubCB(self, ev):
+    def pubCB(self):
         with self.mutex:
             if self.current_msg:
                 self.pub.publish(self.current_msg)
 
-    def checkCB(self, ev):
+    def checkCB(self):
         new_msg = DIAG.DiagnosticArray()
-        new_msg.header.stamp = rospy.get_rostime()
+        # new_msg.header.stamp = rospy.get_rostime()
 
-        st = ntp_diag(self.stat, self.ntp_hostname, self.offset, self.error_offset)
+        st = self.ntp_diag(self.stat, self.ntp_hostname, self.offset, self.error_offset)
```

Comment:
If this is now a class method it does no need all these things passed to it. What it does need would be a more descriptive name, though.

Reply:
Can do. Will add a docstring while I'm at it.

Reply:
Done: 5ee2e11b48903e9621a37c0ff312d9cc7e25527a